### PR TITLE
[ADD] payment_razorpay: integrate with "Razorpay Payment" gateway

### PR DESCRIPTION
**Pull request by mba-odoo**
See README for technical details.

task-2800823

See also:
- https://github.com/odoo/documentation/pull/2149

### CODE REVIEW

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
Hello, I made a first *partial* review pass and highlighted the parts of the code that involved some changes or raised some questions. I will push a fixup commit containing nitpicking changes aimed at cleaning the code only.

I see that you integrated with the Razorpay Standard Checkout solution, but I find that the [Hosted Checkout](https://razorpay.com/docs/payments/payment-gateway/web-integration/hosted) solution would be better suited for our needs:
- The UX would be better than with an in-page modal that badly integrates with Odoo's interface.
- The code would be much simpler because it will allow dropping all the JS customization as well as some Python methods.
- Even if a redirection error occurs, we're covered with the webhook.

Could you please if that is feasible and if it indeed simplifies the code? I will resume the review after we decide on the final solution. :)

File: addons/payment/data/payment_icon_data.xml [hunk 1]
```diff
@@ -151,4 +151,9 @@
         <field name="image" type="base64" file="payment/static/img/barter-by-flutterwave.png"/>
     </record>
 
+    <record id="payment_icon_cc_rupay" model="payment.icon">
+        <field name="sequence">220</field>
```
[the remark this answers is not in the document]
Comment (mba-odoo) [reply addons/payment/data/payment_icon_data.xml:155]: @AntoineVDV , it captures a 60 percent share of India's card market so we prefer to keep it after Paypal at sequence 75. thanks

File: addons/payment/data/payment_acquirer_data.xml [hunk 2]
```diff
@@ -370,6 +370,33 @@
                ])]"/>
     </record>
 
+    <record id="payment_acquirer_razorpay" model="payment.acquirer">
+        <field name="name">Razorpay</field>
```
[the remark this answers is not in the document]
Comment (mba-odoo) [reply addons/payment/data/payment_acquirer_data.xml]: @AntoineVDV , We will display it as `Credit & Debit Card, UPI (Powered by Razorpay)`. Thanks

File: addons/payment/data/payment_icon_data.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (AntoineVDV) [reply addons/payment/data/payment_icon_data.xml]: Fine by me, but let's move it after PayPal so it's less confusing then ;)

**Review by AntoineVDV — APPROVED** [review]
@robodoo override=ci/security
Reason:
- `hmac`, `hashlib`: Used to compute a signature.
- `requests`: Used to make requests to a controlled URL with a short timeout.

Link to ci/security build (master): https://runbot.odoo.com/runbot/batch/846307/build/18940375

@robodoo r+ rebase-ff

**Comment by mba-odoo** [comment]
🥳

**Pull request closed without merging**